DependencyInjection/Compiler/TranslationDomainsExtensionsPass.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformUIBundle\DependencyInjection\Compiler;
+
+use EzSystems\PlatformUIBundle\DependencyInjection\PlatformUIExtension;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class TranslationDomainsExtensionsPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $domains = [];
+        foreach ($container->getExtensions() as $extension) {
+            if ($extension instanceof PlatformUIExtension) {
+                if (isset($hasDuplicates)) {
```

Comment:
I think for now we don't need to handle duplicates. In EZP-26556 we'll rename our own domains with a prefix and we should document clearly that extension developers should do the same. Hopefully this should be enough to avoid duplicated domains.